### PR TITLE
expanding ASP.NET Core 2.0 MVC example app

### PR DESCRIPTION
**Pull request by tremlab**
first pass at the example app.  feedback welcome!

### CODE REVIEW

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs [hunk 1]
```diff
@@ -9,29 +9,71 @@
...
+                    Email = "h.lamarr@ww.com"
+                };
+                //this example makes some modifications that only apply to reports of error class "System.NotImplementedException".
+                if (report.OriginalException is System.NotImplementedException)
+                {
+                    report.Event.Metadata.Add("account", new Dictionary<string, object> { { "paying", true } });
+                    report.Event.Context = "an-important-context";
+                }
+
+                // note that if you return false from the BeforeNotify, this will cancel the entire error report.
```
Comment (martin308) [inline examples/aspnetcore20-mvc/Controllers/HomeController.cs]: this is not true, you need to call `report.Ignore()` to stop the report from sending

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs [hunk 2]
```diff
@@ -9,29 +9,71 @@
...
-    public class HomeController : Controller
+    public class HomeController : Microsoft.AspNetCore.Mvc.Controller
     {
-        private readonly IClient _client;
+        // The Bugsnag client (initialized in Startup.cs) will be injected into your classes where you declare the IClient dependency.
+        // This allows you to report handled exceptions within your controller.
+        private readonly Bugsnag.IClient _bugsnag;
 
-        public HomeController(IClient client)
+        public HomeController(Bugsnag.IClient client)
```
Comment (martin308) [inline examples/aspnetcore20-mvc/Controllers/HomeController.cs]: should this method parameter be renamed to `bugsnag`?

File: examples/aspnetcore20-mvc/README.md [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
+    ```sh
+    git clone https://github.com/bugsnag/bugsnag-dotnet.git
+    cd bugsnag-dotnet/examples/aspnetcore20-mvc
+    ```
+
+1. In the [Startup.cs](Startup.cs) file Replace the `API_KEY` placeholder with your own Bugsnag API key.
 
-In the `Startup.cs` file enter the Bugsnag API key that you want to use with
-this application:
+1. In the [aspnetcore20-mvc.csproj](aspnetcore20-mvc.csproj) file, make sure `Bugsnag.AspNet.Core` is included in your NuGet packages, as well as  `Microsoft.AspNetCore.All`.
```
Comment (martin308) [inline examples/aspnetcore20-mvc/README.md]: Is this an instruction that we want people to follow?

File: examples/aspnetcore20-mvc/README.md [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
 
-```
-services.AddBugsnag(config => {
-    config.ApiKey = "<API_KEY>";
-    config.ProjectNamespaces = new[]{ "aspnetcore11_mvc" };
-    config.ProjectRoots = new[]{ @"C:\app\" };
-});
-```
+1. To report non-fatal exceptions, in the [HomeController.cs](HomeController.cs) file, make sure to declare the `IClient` dependency where you want the Bugsnag client injected into your classes.
```
Comment (martin308) [inline examples/aspnetcore20-mvc/README.md]: is this link to the home controller correct?

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs [hunk 5]
```diff
@@ -9,29 +9,71 @@
...
+            });
+
             return View();
         }
 
-        public IActionResult About()
+        public IActionResult Problems()
         {
-            _client.Breadcrumbs.Leave("Here comes the exception...");
+            // Bugsnag will automatically include the last 25 breabcrumbs to an error report, but you can also manually log breadcrumbs.
```
Comment (martin308) [inline examples/aspnetcore20-mvc/Controllers/HomeController.cs]: is the wording on this correct? Manually logging breadcrumbs is the main way that breadcrumbs are attached to a report but this seems to imply that it's not the main way

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs [hunk 6]
```diff
@@ -9,29 +9,71 @@
...
         public IActionResult Contact()
         {
             ViewData["Message"] = "Your contact page.";
+            try
+            {
+                throw new System.Exception("Error on contact page!");
+            }
+            catch (Exception ex)
+            {
+                // To report handled exceptions, make sure to declare the ICLient dependency (above), then you can pass the exception object to your client for notification.
```
Comment (martin308) [inline examples/aspnetcore20-mvc/Controllers/HomeController.cs]: `ICLient` -> `IClient`

File: examples/aspnetcore20-mvc/README.md [hunk 7]
```diff
@@ -1,27 +1,38 @@
-ASP.NET Core 2.0 MVC Example Application
-====
+# ASP.NET Core 2.0 MVC Example Application
 
-### Requirements
+This example shows how you can use the Bugsnag .NET notifier in a basic ASP.NET CORE 2.0 MVC app. (We also have an [example](https://github.com/bugsnag/bugsnag-dotnet/tree/master/examples/aspnetcore11-mvc) for ASP.NET CORE 1.3)
```
Comment (martin308) [inline examples/aspnetcore20-mvc/README.md]: `CORE` -> `Core`

Comment (tremlab) [reply examples/aspnetcore20-mvc/README.md]: fixed.

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs (hunk 6 above)
Comment (tremlab) [reply examples/aspnetcore20-mvc/Controllers/HomeController.cs]: fixed

File: examples/aspnetcore20-mvc/Controllers/HomeController.cs (hunk 5 above)
Comment (tremlab) [reply examples/aspnetcore20-mvc/Controllers/HomeController.cs]: corrected - thanks for clarification.

File: examples/aspnetcore20-mvc/README.md (hunk 4 above)
Comment (tremlab) [reply examples/aspnetcore20-mvc/README.md]: nope :) fixed.

File: examples/aspnetcore20-mvc/README.md (hunk 3 above)
Comment (tremlab) [reply examples/aspnetcore20-mvc/README.md]: After discussion, came full circle on this.  Yes, will keep instructions on key installation steps, even if they have already been taken care of by the example app.

**Pull request merged**